src/Plugin.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/**
+ * @file
+ * Contains \Netzstrategen\Varnish\Plugin.
+ */
+
+namespace Netzstrategen\Varnish;
+
+/**
+ * Main front-end functionality.
+ */
+class Plugin {
+
+  /**
+   * Prefix for naming.
+   *
+   * @var string
+   */
+  const PREFIX = 'varnish';
+
+  /**
+   * Gettext localization domain.
+   *
+   * @var string
+   */
+  const L10N = self::PREFIX;
+
+  /**
+   * @var string
+   */
+  private static $baseUrl;
+
+  /**
+   * Loads the plugin textdomain.
+   */
+  public static function loadTextdomain() {
+    load_plugin_textdomain(static::L10N, FALSE, static::L10N . '/languages/');
+  }
+
+  /**
+   * @implements init
+   */
+  public static function init() {
+    add_action('save_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('deleted_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('trashed_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('edit_post', __CLASS__ . '::purgePost', 10, 2);
+    add_action('delete_attachment', __CLASS__ . '::purgePost', 10, 2);
+
+    add_action('switch_theme', __CLASS__ . '::purgePost', 10, 2);
+
+    add_action('wp_update_attachment_metadata', __CLASS__ . '::purgeAttachmentMeta', 50, 2);
```

Comment:
Why are you using a custom `camelCase` class method here, while using a `snake_case` method calls [here](https://github.com/netzstrategen/wordpress-varnish/pull/1/files#diff-4fb119e14a4976b590190226e5b0075aR21)?

Reply:
Because multiple hooks are registering the same callback, so there is not a single hook name to use as base for a method name.  The snake_case is used when there is a 1:1 relationship between hook and callback.

Reply:
`purgeAttachmentMeta` is used by `wp_update_attachment_metadata` only. Anyway, I think we can proceed with the current solution. In case another hook makes use of this as well, we don't have to adjust this in multiple places.

Reply:
Ah, I see what you mean now.